packages/app-explorer/src/Events.tsx

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017-2018 @polkadot/app-explorer authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+// Copyright 2017-2018 @polkadot/app-explorer authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { I18nProps } from '@polkadot/ui-app/types';
+
+import React from 'react';
+import { Event, EventRecord } from '@polkadot/types';
+import withObservable from '@polkadot/ui-react-rx/with/observable';
+import withMulti from '@polkadot/ui-react-rx/with/multi';
+
+import { MAX_ITEMS } from './BlockHeaders';
+import translate from './translate';
+
+type Props = I18nProps & {
+  systemEvents?: Array<EventRecord>
+};
+
+type State = {
+  recentEvents: Array<Event>;
+};
+
+class Events extends React.PureComponent<Props, State> {
+  constructor (props: Props) {
+    super(props);
+
+    this.state = {
+      recentEvents: []
+    };
+  }
+
+  static getDerivedStateFromProps ({ systemEvents = [] }: Props, prevState: State): State {
+    const recentEvents = systemEvents
+      .filter(({ event }) =>
+        event.index.toHex() !== '0x0000'
+      )
+      .map(({ event }) =>
+        event
+      )
+      .concat(prevState.recentEvents)
+      .filter((_, index) =>
+          index < MAX_ITEMS
+        );
+
+    return {
+      recentEvents
+    };
+  }
+
+  render () {
+    const { t } = this.props;
+    const { recentEvents } = this.state;
+
+    if (recentEvents.length === 0) {
+      return (
+        <div>{t('events.none', {
+          defaultValue: 'no non-internal events available'
+        })}</div>
+      );
+    }
+
+    return (
+      <pre>{JSON.stringify(recentEvents, null, 2)}</pre>
```

Comment:
had a question about system events in general as I'm thinking about how this event data can be displayed in a more useable fashion. The data is just a number or hex so I'm not sure what I'm looking at when I read `{
    "index": "0x0200",
    "data": [
      7401
    ]
  },` for example.

Reply:
Yeap. This is still WIP - basically the Events has a section + method + arguments (each with a type). Next couple of pushes will start making sense of the data - for now (so I can actually see something), was just dumping the stringified JSON to make sure it does come in.